delphi-checks/src/main/java/au/com/integradev/delphi/checks/EmptyRoutineCheck.java

```diff
@@ -59,4 +76,11 @@ private static boolean shouldAddViolation(RoutineImplementationNode routine) {
         && !declaration.hasDirective(RoutineDirective.VIRTUAL)
         && !InterfaceUtils.implementsMethodOnInterface(declaration);
   }
+
+  private static boolean shouldAddViolation(AnonymousMethodNode anonymousMethod) {
+    if (!anonymousMethod.isEmpty()) {
+      return false;
+    }
+    return anonymousMethod.getStatementBlock().getComments().isEmpty();
```

Comment:
This can be a single statement:
```suggestion
  return anonymousMethod.isEmpty() && anonymousMethod.getStatementBlock().getComments().isEmpty();
```